### PR TITLE
[Core] Update process-factory to work with python-module-only applications

### PR DESCRIPTION
**Pull request by philbucher**
New applications are python-modules-only, they need the new import-mechanism

The changes here are fully backwards-compatible and were also tested with the new python-import mechanism (by @tteschemacher with the IgaApplication that uses the new import-mechanism)

@jcotela this is finally the implementation that we were already discussing

after this is merged I will tackle #3673

### CODE REVIEW

File: kratos/python_scripts/process_factory.py [hunk 1]
```diff
@@ -9,22 +9,36 @@ def ConstructListOfProcesses( self, process_list ):
...
                 if not kratos_module_name.startswith("KratosMultiphysics"):
                     kratos_module_name = "KratosMultiphysics." + kratos_module_name
-                __import__(kratos_module_name)
 
-            # The python_module is the actual script that must be launch
-            if(item.Has("python_module")):
-                python_module = __import__(item["python_module"].GetString())
+                try:
+                    full_module_name = kratos_module_name + "." + python_module_name
+                    python_module = __import__(full_module_name,fromlist=[python_module_name])
```
Comment (philbucher) [inline kratos/python_scripts/process_factory.py:31]: this is the main change essentially

**Comment by philbucher** [comment]
I recommend to review by looking at the entire file, otherwise the changes might not be fully clear

**Review by loumalouomega — DISMISSED** [review]

**Comment by sunethwarna** [comment]
This works :) Thanks @philbucher :)

**Comment by philbucher** [comment]
can this be merged?

**Review by jcotela — APPROVED** [review]

**Pull request merged**